node/worker/worker.go

```diff
@@ -390,7 +390,10 @@ func (w *Worker) FinalizeNewBlock(sig []byte, signers []byte, viewID uint64, coi
 
 	// Shard State
 	if shardState != nil && len(shardState.Shards) != 0 {
-		w.current.header.SetShardStateHash(shardState.Hash())
+		//we store shardstatehash in header only before prestaking epoch (header v0,v1,v2)
+		if !w.config.IsPreStaking(shardState.Epoch) {
```

Comment:
We shouldn't count in the shardState.epoch, it should be the block's epoch. they are not the same thing.

Reply:
shardState.Epoch is the next epoch?